python/tests/integration/arcticdb/test_lmdb.py

```diff
@@ -102,6 +102,30 @@ def test_lmdb_malloc_trim(lmdb_storage):
     lib = ac.create_library("test_lmdb_malloc_trim")
     lib._nvs.trim()
 
+@pytest.mark.skipif(sys.platform != "win32", reason="Non Windows platforms have different file path name restrictions")
+@pytest.mark.parametrize("invalid_lib_name", ["lib?1", "lib:1", "lib|1", "lib.", "lib "])
+def test_invalid_lmdb_lib_name_windows(lmdb_storage, invalid_lib_name):
+    ac = lmdb_storage.create_arctic()
+    with pytest.raises(UserInputException) as e:
+        ac.create_library(invalid_lib_name)
+
+    assert ac.list_libraries() == []
+
+def is_valid_lmdb_lib_name(lmdb_storage, lib_name):
+    ac = lmdb_storage.create_arctic()
+    ac.create_library(lib_name)
+
+    return ac.list_libraries() == [lib_name]
```

Comment:
I'd add this inside the tests instead of separating it in a function. While it helps avoiding code duplication  it has the side effect of actually creating a library. I wouldn't expect a function checking if the name is valid to create a library.